chat_shell/tests/test_truncation_detection.py

```
@@ -0,0 +1,224 @@
+        assert "truncation_reason" not in result
+
+    def test_get_current_result_with_truncation(self, streaming_state):
+        """Test that result includes truncation info when truncated."""
+        streaming_state.append_content("Hello World")
+        streaming_state.set_truncated("max_tokens")
+        result = streaming_state.get_current_result()
+
+        assert result["truncated"] is True
+        assert result["truncation_reason"] == "max_tokens"
+
+    def test_truncation_info_in_result_all_providers(self, streaming_state):
```

Ruff (0.14.14)

[warning] Unused method argument: `streaming_state`

(ARG002)